applications/test_realtime/src/lib.rs

```diff
@@ -0,0 +1,44 @@
+//! This application demonstrates how to create a periodic task using the `sleep::sleep_periodic` API and add it to the realtime scheduler.
+
+#![no_std]
+
+#[macro_use] extern crate log;
+extern crate alloc;
+extern crate spawn;
+extern crate sleep;
+extern crate scheduler;
+
+use core::sync::atomic::AtomicUsize;
+use alloc::{
+    vec::Vec,
+    string::String
+};
+
+pub fn main(_args: Vec<String>) -> isize {
+    if cfg!(realtime_scheduler) {
+        // build and spawn two real time periodic tasks
+        // we will start them as blocked in order to set the periods before they run
+        let periodic_tb1 = spawn::new_task_builder(task_delay_tester, 1).block();
+        let periodic_task_1 = periodic_tb1.spawn().unwrap();        // setting the periods of the tasks
+
+        scheduler::set_periodicity(&periodic_task_1, 1000).unwrap();
+
+        // starting the tasks
+        periodic_task_1.unblock();
+    }
+    else {
+        info!("Realtime scheduler not enabled!");
+    }
+    0
+}    
+
+fn task_delay_tester(arg: usize) {
```

Comment:
Agreed, I am not too familiar with testing capabilities in Rust, but I think a good one to implement could be:

Let the program take in arguments n and p1, ..., pm. For each pi, we spawn a task with period pi, then instead of the logging statement, we put a statement that measures the time elapsed between successive calls and let each of the spawned tasks run for n periods. We can use these measurements to get statistics on how much the time between successive calls of the timing statement deviates from the expected time, i.e. the period pi.

Of course, to achieve this, we would need to have the capability to measure times pretty accurately.

Reply:
ok, if you *want* to add such a test, that's doable using the `hpet` clock source. it's very accurate. 
If you don't want to do this, no worries. Maybe just add a comment at the top of `test_realtime/src/lib.rs` describing what other kinds of tests could/should be done later.